tracer/src/Datadog.Trace/Configuration/GitMetadataTagsProvider.cs

```diff
@@ -80,6 +84,23 @@ public bool TryExtractGitMetadata([NotNullWhen(true)] out GitMetadata? gitMetada
         }
     }
 
+    private void PropagateGitMetadataToTheProfiler(GitMetadata gitMetadata)
+    {
+        try
+        {
+            // Avoid P/Invoke if the profiler is not ready (for obvious reason)
```

Comment:
When does the profiler become ready? Is its initialization a thing that happens asynchronously in the background? Since this method will only get called once in the positive case (and then the results get cached), is it possible that there will be a race condition here where this method is called before the profiler become ready?

Reply:
The profiler is ready when the CLR create its instance and initialize it. So way before managed code started to be executed.

Reply:
👍